pixel/image.go

```diff
@@ -99,7 +108,21 @@ func (img Image[T]) Size() (int, int) {
 func (img Image[T]) setPixel(index int, c T) {
 	var zeroColor T
 
-	if zeroColor.BitsPerPixel()%8 == 0 {
+	switch {
+	case zeroColor.BitsPerPixel() == 1:
+		// Monochrome.
+		x := int16(index) % img.width
+		y := int16(index) / img.width
+		offset := x + (y/8)*img.width
```

Comment:
I don't follow. Isn't the following more efficient, or is there a reason why you picked the above form?

```suggestion
		offset := index / 8
```

(Note that especially divisions are very slow, though divisions by a constant power of two are very fast).

Reply:
At least from my unit testing this change did not pass my unit tests, so I left it as is.